python/pyiceberg/utils/bin_packing.py

```diff
@@ -14,21 +14,54 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+from typing import (
+    Callable,
+    Iterable,
+    List,
+    Optional,
+)
+
+
+class Bin:
+    def __init__(self, target_weight: int) -> None:
+        self.bin_weight = 0
+        self.target_weight = target_weight
+        self.items: List[int] = []
+
+    def weight(self) -> int:
+        return self.bin_weight
+
+    def can_add(self, weight: int) -> bool:
+        return self.bin_weight + weight <= self.target_weight
+
+    def add(self, item: int, weight: int) -> None:
+        self.bin_weight += weight
+        self.items.append(item)
 
 
 class PackingIterator:
-    def __init__(self, items, target_weight, lookback, weight_func, largest_bin_first=False):
+
+    bins: List[Bin]
+
+    def __init__(
+        self,
+        items: Iterable[int],
```

Comment:
I think this needs to be parameterized with `T` rather than `int`. The bins can contain any type and the only thing that matters is that `weight_func` can return a value for them.

Reply:
Got it, I was unsure as well, but all the tests were based around `int`'s. Just updated it